helm/hpcc/templates/thor.yaml

```diff
@@ -30,6 +30,8 @@ data:
       platform:
         type: "thor"
         width: {{ mul ($thorScope.numWorkers | default 1) ( $thorScope.channelsPerWorker | default 1) }}
+      services:
+{{ include "hpcc.generateConfigMapServices" .root | indent 6 }}
```

Comment:
I think this/a services: section may also be needed under:
```
{{ $eclAgentType }}: # hthor or roxie 
```

because this section is going to be used by Thor (manager), but there could be calls to WorkunitServices from the job agent (outside of Thor).

Did you hit the Thor scenario with some test ECL code?

Reply:
Is it under the {{ $eclAgentType }}: # hthor or roxie  now? In generated /etc/config/thor.yaml, the services section is under the hthor.
 
I tested hthor, roxie-workunit, and thor using:
OUTPUT(STD.System.Workunit.WorkunitExists('W20210522-211941', false, true));

Reply:
That ECL won't actually run a Thor/in Thor or Roxie, because it's too simple.
It will instead generate that call in line and run in the eclagent.

You'll need to call inside an activity to test, from an APPLY would work I think.

Reply:
Now, I learn how to force some ECL code running on thor (not roxie yet).
You are correct. The fix does not work for thor. I got the error: System error: 20069: Graph graph1[1], SLAVE #1 [10.1.8.130:20100]: Sasha 'sasha-wu-archiver' server not found.

Reply:
The system error (20069: Graph graph1[1], SLAVE #1 [10.1.8.130:20100]: Sasha 'sasha-wu-archiver' server not found) is 'from' eclagent. My debug code shows that the getComponentConfigSP() returns empty when the error is generated. I am not sure how that happens. For this case, which hpcc component or program (ECL WU dll?) loads the workunitservices plugin (with no ComponentConfig)? @jakesmith @ghalliday any suggestion?

Reply:
I can look again, but I think it's because the services: are not being generated into all the component configuration sections that need them, i.e. under thor: in this case.

Reply:
I can understand that the getComponentConfigSP() here does not return the services. But, why does it return nothing (no services, and no other item like analyzeWorkunit="true" config="/etc/config/thor.yaml" daliServers="mydali")?

Reply:
Using the same test ecl code (with the Apply and logical file) on roxie, no additional pod is generated. My test code shows  the getSashaNodes() is called successfully inside the log of the roxie-workunit pod (same as using the simple OUTPUT ecl). Not sure other ECL test I should use to test roxie.